api/src/main/java/org/pmiops/workbench/profile/ProfileService.java

```diff
@@ -381,47 +395,72 @@ private void validateAreaOfResearch(Profile profile) throws BadRequestException
    * object.
    *
    * @param updatedProfile
-   * @param prevProfile
+   * @param previousProfile
    * @throws BadRequestException
    */
   @VisibleForTesting
-  public void validateProfile(@Nonnull Profile updatedProfile, @Nullable Profile prevProfile)
+  public void validateProfile(@Nonnull Profile updatedProfile, @Nullable Profile previousProfile) {
+
+    final boolean isNewProfile = previousProfile == null;
+    final Diff diff = javers.compare(previousProfile, updatedProfile);
+
+    validateProfileForCorrectness(updatedProfile, isNewProfile, diff);
+
+    if (userProvider.get().hasAuthority(Authority.ACCESS_CONTROL_ADMIN)) {
+      validateChangesAllowedByAdmin(diff);
+    } else {
+      validateChangesAllowedByUser(diff);
+    }
+  }
+
+  private void validateProfileForCorrectness(Profile profile, boolean isNewProfile, Diff diff)
```

Comment:
I don't like the `isNewProfile` flag here, because the Diff should already be able to tell us that. E.g. something like `boolean isNewObject(Diff diff)`. Then each validate method could just take in a Diff and a string for the field name, or we could have a `validateFields(Diff diff, Set<String> fieldNames)`. there are lots of `if` statements here currently, which are easily breakable by maintainers.

Reply:
ok fine: isNewProfile is now a function of the Diff.

I'm keeping the ifs because I like how they make the intent explicit.